src/controllers/update/overseas.entity.query.controller.ts

```diff
@@ -14,25 +17,54 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
     return res.render(config.OVERSEAS_ENTITY_QUERY_PAGE, {
       backLinkUrl: config.UPDATE_INTERRUPT_CARD_URL,
       templateName: config.OVERSEAS_ENTITY_QUERY_PAGE,
-      [OeNumberKey]: appData?.[OeNumberKey],
+      [EntityNumberKey]: appData[EntityNumberKey],
     });
   } catch (error) {
     logger.errorRequest(req, error);
     next(error);
   }
 };
 
-export const post = (req: Request, res: Response, next: NextFunction) => {
+export const post = async (req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `POST ${config.OVERSEAS_ENTITY_QUERY_PAGE}`);
-    const oeNumber = req.body[OeNumberKey];
 
-    setExtraData(req.session, { ...getApplicationData(req.session), [OeNumberKey]: oeNumber });
-    return res.redirect(config.UPDATE_OVERSEAS_ENTITY_CONFIRM_URL);
+    const entityNumber = req.body[EntityNumberKey];
```

Comment:
maybe we can add getCompanyProfile and creating the error into validation.middleware(separate method), so that the error is created in validation.middleware, not sure if its right to call companyProfile from validation.middleware

Reply:
I did try this approach but found that the middleware uses a different request type (in meta data passed in) so could not call out to get the company profile. Also, it seems wrong to call out to services in the validation middleware which presumably should just be checking format and other data integrity / validations.

Reply:
Also this avoids calling getProfile twice which itself has downsides

Reply:
> I did try this approach but found that the middleware uses a different request type (in meta data passed in) so could not call out to get the company profile. Also, it seems wrong to call out to services in the validation middleware which presumably should just be checking format and other data integrity / validations.

I agree -- Validation middleware should just do the validations 👍